src/sql/parts/grid/views/editData/editData.component.ts

```diff
@@ -458,27 +465,34 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 	// Adds an extra row to the end of slickgrid (just for rendering purposes)
 	// Then sets the focused call afterwards
 	private addRow(row: number, column: number): void {
-		// Add a new row to the edit session in the tools service
-		this.dataService.createRow();
-
-		// Adding an extra row for 'new row' functionality
-		this.dataSet.totalRows++;
-		this.dataSet.maxHeight = this.getMaxHeight(this.dataSet.totalRows);
-		this.dataSet.minHeight = this.getMinHeight(this.dataSet.totalRows);
-		this.dataSet.dataRows = new VirtualizedCollection(
-			this.windowSize,
-			this.dataSet.totalRows,
-			this.loadDataFunction,
-			index => { return { values: [] }; }
-		);
-
-		// Refresh grid
-		this.onScroll(0);
+		let self = this;
```

Comment:
you shouldn't need to use `self` variable..TypeScript auto defines a `this` alias inside `=>` functions.  And using `this` is the preferred TypeScript pattern.